### ISSUE TITLE
Doc nicety: use * to indicate when arguments must be keywords

### ISSUE THREAD

**Issue by girving** [tensorflow/tensorflow#13165]
https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits shows the signature

    softmax_cross_entropy_with_logits(
        _sentinel=None,
        labels=None,
        logits=None,
        dim=-1,
        name=None
    )

In Python 3, the syntax for `_sentinel=None` is just `*`, and the signature would be

    softmax_cross_entropy_with_logits(
        *,
        labels,
        logits,
        dim=-1,
        name=None
    )

where I've additionally removed the false defaults from labels and logits.  This can't be done to the code which has to be 2.7 compatible, but it could be done to the documentation.

**Comment by girving** [comment]
I'm fine if this gets closed, but probably won't work on it personally.
